api/src/main/java/org/openmrs/validator/DrugOrderValidator.java

```diff
@@ -108,16 +109,30 @@ public void validate(Object obj, Errors errors) {
 			validateFieldsForOutpatientCareSettingType(order, errors);
 			validatePairedFields(order, errors);
 			validateUnitsAreAmongAllowedConcepts(errors, order);
-			validateForRequireDrug(errors);
+			validateForRequireDrug(errors, order);
 		}
 	}
-	
-	private void validateForRequireDrug(Errors errors) {
+
+	private void validateForRequireDrug(Errors errors, DrugOrder order) {
 		//Reject if global property is set to specify a formulation for drug order
 		boolean requireDrug = Context.getAdministrationService().getGlobalPropertyValue(
-		    OpenmrsConstants.GLOBAL_PROPERTY_DRUG_ORDER_REQUIRE_DRUG, false);
-		if (requireDrug) {
-			ValidationUtils.rejectIfEmpty(errors, "drug", "DrugOrder.error.drugIsRequired");
+				OpenmrsConstants.GLOBAL_PROPERTY_DRUG_ORDER_REQUIRE_DRUG, false);
+		OrderService orderService = Context.getOrderService();
+
```

Comment:
We are first checking if drug is required.. 
Based on concept DRUG OTHER and drug further validations will be done..